packages/sources/blocksize-capital-test/src/endpoint/price.ts

```diff
@@ -0,0 +1,138 @@
+  base: {
+    aliases: ['from', 'coin'],
+    type: 'string',
+    description: 'The symbol of symbols of the currency to query',
+    required: false,
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+    required: false,
+  },
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: { base: string; quote: string }
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: Message[]
+  }
+}
+
+type WebsocketTransportGenerics = TransportGenerics & {
+  Provider: {
+    WsMessage: unknown
+  }
+}
+
+export class BlocksizeWebsocketReverseMappingTransport<
+  T extends WebsocketTransportGenerics,
+  K,
+> extends WebsocketReverseMappingTransport<T, K> {
+  api_key = ''
+}
+
+export const websocketTransport: BlocksizeWebsocketReverseMappingTransport<EndpointTypes, string> =
+  new BlocksizeWebsocketReverseMappingTransport<EndpointTypes, string>({
+    url: ({ adapterConfig: { WS_API_ENDPOINT, API_KEY } }) => {
+      websocketTransport.api_key = API_KEY
+      return WS_API_ENDPOINT
+    },
+    handlers: {
+      open: (connection) => {
+        connection.send({
+          jsonrpc: '2.0',
+          method: 'authentication_logon',
+          params: { api_key: websocketTransport.api_key },
+        })
+      },
+      message: (message) => {
+        if (Object.keys(message).length === 0) {
+          logger.debug('WS message is empty, skipping')
+          return []
+        }
+        const [_, msg] = message
+        if (!(msg.method === 'vwap' || 'method' in msg)) return []
+        const [updates] = msg.params.updates
+        const params = websocketTransport.getReverseMapping(updates.ticker)
+        if (!params) {
+          return undefined
+        }
+        const base = params.base
+        const quote = params.quote
+        return [
+          {
+            params: { base, quote },
+            response: {
+              result: updates.price as number,
+              data: {
+                result: updates.price as number,
+              },
+              timestamps: {
+                providerIndicatedTimeUnixMs: new Date(Date.now()).getTime(),
+              },
+            },
+          },
+        ]
+      },
+    },
+    builders: {
+      subscribeMessage: (params) => {
+        const pair = `${params.base}${params.quote}`
+        websocketTransport.setReverseMapping(pair, params)
+        return {
+          jsonrpc: '2.0',
+          method: 'vwap_subscribe',
+          params: { tickers: [pair] },
+        }
+      },
+
+      unsubscribeMessage: (params) => {
+        const pair = `${params.base}${params.quote}`
+        return {
+          jsonrpc: '2.0',
+          method: 'vwap_unsubscribe',
+          params: { tickers: [pair] },
+        }
+      },
+    },
+  })
+
+export const endpoint = new PriceEndpoint<EndpointTypes>({
+  name: 'price',
```

Comment:
Adding this comment so we don't forget but we should update this to the `CryptoPriceEndpoint` once that's released with the framework. Otherwise we'd need to add an alias for `crypto` here.

Reply:
Done